app/src/main/java/com/hedvig/app/feature/offer/ui/checkout/CheckoutViewModel.kt

```diff
@@ -0,0 +1,75 @@
+package com.hedvig.app.feature.offer.ui.checkout
+
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import com.hedvig.app.util.ValidationResult
+import com.hedvig.app.util.validateEmail
+import com.hedvig.app.util.validateNationalIdentityNumber
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.StateFlow
+import kotlinx.coroutines.flow.combine
+import kotlinx.coroutines.flow.debounce
+import kotlinx.coroutines.flow.launchIn
+
+private const val INPUT_DELAY_MS = 1500L
+
+class CheckoutViewModel : ViewModel() {
+
+    protected val _viewState = MutableStateFlow<ViewState>(
+        ViewState.Input(
+            emailInputState = ViewState.InputState.NoInput,
+            identityInputState = ViewState.InputState.NoInput
+        )
+    )
+    val viewState: StateFlow<ViewState> = _viewState
+
+    private val emailInput = MutableStateFlow("")
+    private val identityNumberInput = MutableStateFlow("")
+
+    init {
+        emailInput.debounce(INPUT_DELAY_MS)
```

Comment:
This can probably be prettier. But I realised now that this kind of validation is not that UX friendly? Perhaps its better to validate once the user press the "Get Hedvig"-button. I'll ask Hiro.